### DISCUSSION TITLE
jsonable_encoder on user object that contains a WKBElement

### DISCUSSION THREAD

**Discussion by OddBlueDog** [fastapi/fastapi#7041] [Questions] [answered]
How do I get jsonable_encoder to work with a user object that contains a WKBElement (geography point) ?

This is a `print(vars(current_user))`:

`{'_sa_instance_state': <sqlalchemy.orm.state.InstanceState object at 0x7f2e16dbf590>, 'user_name': 'test', 'location': <WKBElement at 0x7f2e16dbf6d0; 010100000000000000000000400000000000000840>}`

When I do `jsonable_encoder(current_user)` I get: 

`ValueError: [ValueError('dictionary update sequence element #0 has length 1; 2 is required'), TypeError('vars() argument must have __dict__ attribute')]`

**Comment by Mause** [comment]
In what format are you expecting the WKBElement to be serialised?

**Comment by OddBlueDog** [comment, marked as answer]
> In what format are you expecting the WKBElement to be serialised?

Thanks I got it working OK now with the custom_encoder on jsonable_encoder, took me awhile to figure out how to use it. I don't actually need the value at this point so I'm just setting it to None now like ` jsonable_encoder(current_user, custom_encoder={WKBElement: lambda:  None} )`

**Comment by tiangolo** [comment]
Thanks for the help here @Mause ! :clap: :bow: 

Thanks for reporting back and closing the issue @OddBlueDog  :+1:

**Comment by jorenvs** [comment]
I know the issue has been fixed, but I just hit upon the same thing and it took me a while to end up here and figure out the WKBElement was the issue (I had a JSON column with a dict that I thought was causing this). 

My question is mainly: is there some way the error can be made a bit more clear? As in, point out that the JSON serialisation is failing and on which field/value?

Especially for newcomers to FastAPI (read: me) it's not directly evident where the jsonable_encoder is being called (I thought it was in the JSON update step in SQLAlchemy). The stack trace wasn't much help either.
